### PR TITLE
Full API documentation

### PR DESCRIPTION
**Pull request by r-scheele**
- Added introduction section to explain what the API is about
- Added an outline for easy accessibility of the docs
- Added the API architecture and also provided a link to view on Figma
- Provided a detailed explanation of each commands with examples: what they are and how to use them.
- The delete policy needs to identify individual policy so it doesn’t affect the wrong policy. with the way our app works now, if we have an existing rules before connecting to the API, the app won't recognize individual policies, which is why it deletes the entire file if we send a delete request. i want us to be able delete just the affected policy and not all. 
Approach:
check the database and delete the policy specified, and then query the database again to get the list of policies remaining for that user after deleting the one specified, then write them again

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,3 +1,17 @@
...
+
+ADD . .
+
+RUN pip install -r requirements.txt
+
+EXPOSE 8080
+
+COPY . /app
+
+=======
```
Comment (Youngestdev) [inline Dockerfile:14]: Fix merge conflicts.

File: Dockerfile [hunk 2]
```diff
@@ -7,4 +21,5 @@ RUN pip3 install poetry
 RUN poetry config virtualenvs.create false
 RUN poetry install --no-dev
 COPY . .
+>>>>>>> origin/main
```
Comment (Youngestdev) [inline Dockerfile:24]: fix conflict.

File: docs.md [hunk 3]
```diff
@@ -1,23 +1,216 @@
...
+   ```
+- Pull the official image from Docker Hub - and start the container 
+
+   ```bash
+   $ docker pull rscheele3214/rego-builder:latest
+   $ docker run rscheele3214/rego-builder:latest
+   ```
+
+
+Usage
```
Comment (Youngestdev) [inline docs.md:69]: Before directly going into usage, I'll expect to be briefed on how the translation from JSON to REGO works. I have no idea of commands - what do they stand for as well?

I'll expect a brief/detailed section of what commands are - you can then list the various commands here.

After explaining what the commands are, you should give us a lead-in on how they are compiled to form a rule - command, and properties. There's no explanation of what properties are.

**Review by Youngestdev — CHANGES_REQUESTED** [review]
Good job. Suggested some extra information.

File: docs.md (hunk 3 above)
Comment (r-scheele) [reply docs.md:69]: Alright would submit another commit for review soon

File: docs.md [hunk 4]
```diff
@@ -1,23 +1,396 @@
...
-}`
--  Handling equality check between a property on the request object and a parameter
\ No newline at end of file
+View the Architecture in the browser [Here](https://www.figma.com/file/684S7kO4dPQbbZFZr6xZOn/Rego-builder?node-id=0%3A1) 
+
+
+
+   Installation
+   ============
+- Manual installation - Available [Here](https://github.com/r-scheele/rego_builder#readme)
```
Comment (Youngestdev) [inline docs.md:55]: Instead of directing users to another document, place the steps here as well.

File: docs.md [hunk 5]
```diff
@@ -1,23 +1,396 @@
...
+
+- Build your own image and start the container -
```
Comment (Youngestdev) [inline docs.md:57]:
```suggestion
- Build the application image and start the container:
```

File: docs.md [hunk 6]
```diff
@@ -1,23 +1,396 @@
...
+   ```bash
```
Comment (Youngestdev) [inline docs.md:58]:
```suggestion
   ```console
```

File: docs.md [hunk 7]
```diff
@@ -1,23 +1,396 @@
...
+   $ Configure environment variables - the .env file
+   $ docker-compose -f docker-compose.dev.yml up -d
+
+   ```
+- Pull the official image from Docker Hub - and start the container
```
Comment (Youngestdev) [inline docs.md:63]:
```suggestion
- Pull the official image from Docker Hub and start the container 
```

File: docs.md [hunk 8]
```diff
@@ -1,23 +1,396 @@
...
+
+- Build your own image and start the container -
+   ```bash
+   $ Configure environment variables - the .env file
+   $ docker-compose -f docker-compose.dev.yml up -d
+
+   ```
+- Pull the official image from Docker Hub - and start the container 
+
+   ```bash
```
Comment (Youngestdev) [inline docs.md:65]:
```suggestion
   ```console
```

File: docs.md [hunk 9]
```diff
@@ -1,23 +1,396 @@
...
+         "value": "GET"
+      }
+   }
+   ```
+In this case, The command key represents the operation to be performed and the properties key represents the properties that are being used in the operation.
+`input_prop_equals` is the command in example, which initiates the appropriate operation, on the properties object.
+The above rule translates to an equality check between the input_property and the value. <br />
+The REGO equivalent of the above rule object is: <br />
+   `input.request_method == "GET"`
+
```
Comment (Youngestdev) [inline docs.md:89]: You should start this section with a lead like this:

A rule is defined by two keys: command and properties. The command key holds one of the recognized commands and the properties hold another dictionary containing the `input_property` and `value`. In special cases, the `datasource_items` items are also used/added etc etc.

File: docs.md [hunk 10]
```diff
@@ -1,23 +1,396 @@
...
+### As of now, the API supports the following commands:
```
Comment (Youngestdev) [inline docs.md:90]: No need for this to be a heading.

File: docs.md [hunk 11]
```diff
@@ -1,23 +1,396 @@
...
+1. input_prop_equals
+2. input_prop_in
+3. input_prop_in_as
+4. allow_full_access
+
+
+Usage
+============
+### Detailed explanation of commands with examples: 
+
```
Comment (Youngestdev) [inline docs.md:100]: Add a sentence or two explaining what this section will contain.

File: docs.md [hunk 12]
```diff
@@ -1,23 +1,396 @@
...
+
+## 1. Input_props_equals
```
Comment (Youngestdev) [inline docs.md:102]: Preferably, don't number them.

File: docs.md [hunk 13]
```diff
@@ -1,23 +1,396 @@
...
+   }` <br />  and the rule object is:
+   ```json     
+   {
+      "command": "input_prop_equals",
+      "properties": {
+         "input_property": "request_path",
+         "value": ["v1", "collections", "*"]
+      }
+   }
+   ```
```
Comment (Youngestdev) [inline docs.md:120]: Rephrase.

Example:

This logic handles all the paths after a particular section. Let's look at an example rule:

---json file---

In the JSON above, the `value` key contains an asterik in the values section. This indicates that the endpoint x is allowed to do y.

---rego equivalent---

Same applies for every other command in this section.

File: docs.md [hunk 14]
```diff
@@ -1,23 +1,396 @@
...
+============
+This API is responsible for converting JSON variables to REGO which in turn is used in application policy management. The policy management application is born from the need to manage Open Policy Agent ( OPA ) policies written in REGO from a user interface for non-technical users and an API through JSON for technical users.
+
+  Features
+============
+   - Translating JSON to REGO
+   - CRUD operations for policies for persistence.
+   - Supports existing policies for editing.
+   - Supports creating new policies from scratch.
+
```
Comment (Youngestdev) [inline docs.md:40]: Repetition of the second item on the list.

File: docs.md [hunk 15]
```diff
@@ -1,23 +1,396 @@
...
+
+   API CRUD Operations
+   ============
+
+   The API is designed to support all CRUD operations that makes interaction with the policies easy. The following routes are currently defined:
+
+ ###  - POST `/policies/`  🗝
+Using the JSON defined from the frontend, this route is used to build a new policy. The response will be a REGO file written and pushed to github as a newly established remote repository, with the request body conforming to a specified syntax described by the pydantic model `Policy`. <br />
+```py3
```
Comment (Youngestdev) [inline docs.md:252]: Instead:

The `/policies` route receives a JSON request body comprising the rules needed to create a new policy. The policy upon a successful operation is created and the defined GitHub repository is updated with the policy file of _.rego_ format.

Anyone can choose to use the API and not necessarily a frontend..

File: docs.md [hunk 16]
```diff
@@ -1,23 +1,396 @@
...
+    command: str
+    properties: Dict[str, Union[str, List[str], Dict[str, Union[str, List[str]]]]]
+```
+
+Each Rule object forms a specific rule in a Allow block, and a list of Rules forms a Allow block. <br />
+```py3
+List[Rule] == Allow {
+   ...
+}
+```
```
Comment (Youngestdev) [inline docs.md:269]: The definition of the rule object should be before the explanations of the path operations.

File: docs.md [hunk 17]
```diff
@@ -1,23 +1,396 @@
...
+A typical request body which will contain multiple allow blocks in the REGO rule would be: <br />
```
[the remark this answers is not in the document]
Comment (Youngestdev) [reply docs.md:270]: Also, can we make the example rule shorter?

File: docs.md [hunk 18]
```diff
@@ -1,23 +1,396 @@
...
+      ]
+   ]}
+
+   ```
+ ###  - GET `/policies/` 🗝
+This route is used to get all policies that have been created. The response will be a list of all policies that have been created by a certain user, and contains all the associating rules with the policy <br />
+
+```py3
+List[Policy]   
+   ```
```
Comment (Youngestdev) [inline docs.md:388]: Not needed here.

File: docs.md [hunk 19]
```diff
@@ -1,23 +1,396 @@
...
+
+   ###  - GET PUT DELETE `/policies/{policy_name}` 🗝
+   GET - This request method is used to get a specific policy by name. The response will be a policy object conforming to the pydantic model `Policy`. <br />
+   PUT - This request method is used to update a specific policy by name. The response will be a policy object conforming to the pydantic model `Policy`. <br />
+   DELETE - This request method is used to delete a specific policy by name
```
Comment (Youngestdev) [inline docs.md:393]: They all need to show:

- Sample requests and responses
- What happens when a policy isn't found?

**Review by Youngestdev — CHANGES_REQUESTED** [review]
Great job, Habeeb. I left some comments.

File: docs.md [hunk 20]
```diff
@@ -1,23 +1,476 @@
...
+      "properties": {
+         "input_property": "request_path",
+         "value": ["v1", "collections", "*"]
+      }
+   }
+   ```
+   In the json above, the `value` key holds an asterik in the values section, to indicate that the endpoint x is allowed to do y.
+
+   The REGO equivalent of the above rule object is: <br />
+   `input.request_path[0] == "v1" && input.request_path[1] == "collections"`
```
Comment (Youngestdev) [inline docs.md:167]: Put this in a code block.

File: docs.md [hunk 21]
```diff
@@ -1,23 +1,476 @@
...
+       {
+         "command": "input_prop_equals",
+         "properties": {
+            "input_property": "request_path",
+            "value": "admin",
+         },
+      }
+   ]
+   ```
+In the json above, the list of objects indicate a very speciall `allow` block, that combines two commands. The `input_property` key holds the property that is to be va;idated before the request to that path is passed.
```
Comment (Youngestdev) [inline docs.md:232]:
```suggestion
In the JSON above, the list of objects indicates a very special `allow` block, that combines two commands. The `input_property` key holds the property that is to be validated before the request to that path is passed. 
```

File: docs.md [hunk 22]
```diff
@@ -1,23 +1,476 @@
...
+            "value": "admin",
+         },
+      }
+   ]
+   ```
+In the json above, the list of objects indicate a very speciall `allow` block, that combines two commands. The `input_property` key holds the property that is to be va;idated before the request to that path is passed. 
+
+The REGO equivalent of the above rule object is: <br />
+   `allow {
+  input.request_path == ["v1", "collections", "lakes", ""] && input.groupname == "admin"
```
Comment (Youngestdev) [inline docs.md:236]: indent properly.

File: docs.md [hunk 23]
```diff
@@ -1,23 +1,476 @@
...
+}` 
+
+   ## 2. Input_props_in
```
Comment (Youngestdev) [inline docs.md:239]: Remove number and it can't start with a block letter.

File: docs.md [hunk 24]
```diff
@@ -1,23 +1,476 @@
...
+   {
+      "command": "input_prop_in",
+      "properties": {
+         "input_property": "company",
+         "datasource_name": "items",
+         "datasource_loop_variable": "name"
+      }
+}
+   ```
+   In the json above, the `input_property` key holds the property that is to be validated before the request to that path is passed. The `datasource_name` key holds the name of the datasource(a list) from the database. The `datasource_loop_variable` key holds the name of the key on each object in the datasource.
```
Comment (Youngestdev) [inline docs.md:254]: Might be good to give a sentence or two about what data source items are and how they're gotten - from a database or smtn.

File: docs.md [hunk 25]
```diff
@@ -1,23 +1,476 @@
...
+
+   The REGO equivalent of the above rule object is: <br />
+   `input.company == data.items[i].name`
+
+   ## 3. Input_props_in_as
```
Comment (Youngestdev) [inline docs.md:259]: Remove number and word shouldn't start with a block letter.

File: docs.md [hunk 26]
```diff
@@ -1,23 +1,476 @@
...
+         "input_properties": ["preferred_username", "groupname"]
+      }
+ }
+   ```
+   In the json above, the resulting REGO code loops over the datasource twice, checking for equality between the values of the input properties, and the values of the datasource loop variables.
+
+   The REGO equivalent of the above rule object is: <br />
+         `some i` <br />
+         `data.items[i].name == input.preferred_username` <br />
+         `data.items[i].everyone == input.groupname` <br />
```
Comment (Youngestdev) [inline docs.md:279]: Place it in a codeblock.

File: docs.md [hunk 27]
```diff
@@ -1,23 +1,476 @@
...
+   In the json above, the resulting REGO code loops over the datasource twice, checking for equality between the values of the input properties, and the values of the datasource loop variables.
+
+   The REGO equivalent of the above rule object is: <br />
+         `some i` <br />
+         `data.items[i].name == input.preferred_username` <br />
+         `data.items[i].everyone == input.groupname` <br /> 
+
+
+
+## 4. allow_full_access
```
Comment (Youngestdev) [inline docs.md:283]: Remove number.

File: docs.md [hunk 28]
```diff
@@ -1,23 +1,476 @@
...
+   This logic allows full access to the resources defined. if the value of the property on the input object has a particular value<br />
```
Comment (Youngestdev) [inline docs.md:284]:
```suggestion
   This logic allows full access to the resources defined. If the value of the property on the input object has a particular value<br />
```

File: docs.md [hunk 29]
```diff
@@ -1,23 +1,476 @@
...
+      {
+         "command": "allow_full_access",
+         "properties": {
+            "input_property": "groupname",
+            "value": "EDITOR_ATAC",
+         },
+      }
+   ]
+   ```
+   In the json above, the result is an allow block, that allows access to the resources defined, if the value of the property on the input object has a particular value.
```
Comment (Youngestdev) [inline docs.md:298]:
```suggestion
   In the JSON above, the result is an allow block that allows access to the resources defined if the value of the property on the input object has a particular value.
```

File: docs.md [hunk 30]
```diff
@@ -1,23 +1,476 @@
...
+
+
+
+   API CRUD Operations
+   ============
+
+   The API is designed to support all CRUD operations that makes interaction with the policies easy. The following routes are currently defined:
+
+ ###  - POST `/policies/`  🗝
+Using the JSON defined from the frontend, this route is used to build a new policy. The response will be a REGO file written and pushed to github as a newly established remote repository, with the request body conforming to a specified syntax described by the pydantic model `Policy`. <br />
```
Comment (Youngestdev) [inline docs.md:312]:
```suggestion
Using the JSON defined from the frontend, this route is used to build a new policy. The response will be a REGO file written and pushed to GitHub as a newly established remote repository, with the request body conforming to a specified syntax described by the pydantic model `Policy`. <br />
```

File: docs.md [hunk 31]
```diff
@@ -1,23 +1,476 @@
...
+   {"status": 200, "message": "Policy created successfully"}
+   ```
+   ###  - GET `/policies/` 🗝
+   This route is used to get all policies that have been created. The response will be a list of all policies that have been created by a certain user, and contains all the associating rules with the policy <br />
+
+   An example response body is: <br />
+   ```json
+   [
+      {
+         "name": "Example",
```
Comment (Youngestdev) [inline docs.md:373]: The name should match the one in the POST request sent above.

File: docs.md [hunk 32]
```diff
@@ -1,23 +1,476 @@
...
+                  "input_property": "groupname",
+                  "value": "EDITOR_ATAC"
+               }
+            }
+         ]
+      }
+   ]
+   ```
+
+   ###  - GET PUT DELETE `/policies/{policy_name}` 🗝
```
Comment (Youngestdev) [inline docs.md:417]: Can you put each operation in a different section as opposed to this?

**Review by Youngestdev — CHANGES_REQUESTED** [review]
Great job. I left a few comments as well as suggestions. Once these have been attended to, request a review so I can check again.

Excellent work!

File: docker-compose.prod.yml [hunk 33]
```diff
@@ -2,7 +2,7 @@ version: "3"
 
 services:
   web:
-    build: .
+    image: rscheele3214/rego_builder
```
Comment (Youngestdev) [inline docker-compose.prod.yml:5]:
```suggestion
    build: .
```

File: app/utils/write_rego.py [hunk 34]
```diff
@@ -25,33 +25,20 @@ def write_to_file(self, rule: dict, operation: str = "write") -> None:
...
         file_path = f"{self.github.local_repo_path}/auth.rego"
 
         if not os.path.exists(file_path):
             raise FileNotFoundError
 
-        file = open(file_path, "w")
-        file.close()
-
-        # Update GitHub
-        self.github.push()
```
Comment (Youngestdev) [inline app/utils/write_rego.py:56]: It doesn't update GitHub?

**Review by Youngestdev — CHANGES_REQUESTED** [review]
Doesn't effect any of the changes to GitHub.

File: app/utils/write_rego.py (hunk 34 above)
Comment (r-scheele) [reply app/utils/write_rego.py:56]: It does. Not sure this is my pr

Comment (r-scheele) [reply app/utils/write_rego.py:56]: Check the last commit

Comment (Youngestdev) [reply app/utils/write_rego.py:56]: hm? ok.

File: .github/workflows/main.yml [hunk 35]
```diff
@@ -20,23 +20,13 @@ jobs:
...
-        import os
-        import pip
-        with open(os.environ["GITHUB_ENV"], mode="a") as io:
-            print(f"VIRTUALENV_PIP={pip.__version__}", file=io)
-
-
-    - name: Format code with Black
-      uses: rickstaa/action-black@v1
-      with:
-        black_args: "."
```
Comment (Youngestdev) [inline .github/workflows/main.yml]: Why was this removed? 🤔

Comment (r-scheele) [reply .github/workflows/main.yml]: I noticed it doesn't work🤔 or does it?

File: docs.md [hunk 36]
```diff
@@ -1,23 +1,499 @@
...
-   path = /collections/lakes
-}`
--  Handling equality check between a property on the request object and a parameter
\ No newline at end of file
+View the Architecture in the browser [Here](https://www.figma.com/file/684S7kO4dPQbbZFZr6xZOn/Rego-builder?node-id=0%3A1) 
+
+
+
+   Installation
+   ============
```
Comment (Youngestdev) [inline docs.md:53]: We can add a line to tell users that there are a number of ways to install the application.

File: docs.md [hunk 37]
```diff
@@ -1,23 +1,499 @@
...
+GITHUB_USERNAME=<your_github_username>
+GITHUB_URL=<your_github_url where the authorization code lives>
+CLIENT_ID=<your_github_client_id>
+CLIENT_SECRET=<your_github_client_secret>
+SECRET_KEY=<your_secret_key>
+ALGORITHM=<your_algorithm e.g HS256>
+ENVIRONMENT=<your_environment e.g. production|development>
+```
+
+- Run the application from the entry point
```
Comment (Youngestdev) [inline docs.md:76]:
```suggestion
- Run the application from the entry point:
```

File: docs.md [hunk 38]
```diff
@@ -1,23 +1,499 @@
...
+   $ Configure environment variables - the .env file
+   $ docker-compose -f docker-compose.dev.yml up -d
+
+   ```
+### - Pull the official image from Docker Hub and start the container 
+
+   ```console
+   $ docker pull rscheele3214/rego-builder:latest
+   $ docker run rscheele3214/rego-builder:latest
+   ```
```
Comment (Youngestdev) [inline docs.md:94]: Why are there indentations in the document?

File: docs.md [hunk 39]
```diff
@@ -1,23 +1,499 @@
...
+List[Rule] == Allow {
+   ...
+}
+```
+
+The API supports the following commands; input_prop_equals, input_prop_in, input_prop_in_as, allow_full_access.
+
+Usage
+============
+### Detailed explanation of commands with examples:
```
Comment (Youngestdev) [inline docs.md:145]: I think we can add a sentence about what this section entails instead of presenting another heading.

File: docs.md [hunk 40]
```diff
@@ -1,23 +1,499 @@
...
+
+Usage
+============
+### Detailed explanation of commands with examples: 
+This section contains detailed explanations of the commands and examples of how to use them. 
+### input_props_equals
+   This command has different logic to handle series of equality checks.
+  - Handling '*' as the wildcard flag: <br /> 
+
+   This logic handles all the paths after a particular section. if `/collections/` is supplied as the option, all the routes after it will be allowed e.g allow `/collections/obs/`, allow `/collections/test-data/obs/`, allow `/collections/obs/` etc. <br />
```
Comment (Youngestdev) [inline docs.md:151]: Remove indents.

File: docs.md [hunk 41]
```diff
@@ -1,23 +1,499 @@
...
+   Example:
+   ```json     
+   {
+      "command": "input_prop_equals",
+      "properties": {
+         "input_property": "request_path",
+         "value": ["v1", "collections", "*"]
+      }
+   }
+   ```
```
Comment (Youngestdev) [inline docs.md:162]: Remove indents.

File: docs.md [hunk 42]
```diff
@@ -1,23 +1,499 @@
...
+      "properties": {
+         "input_property": "company",
+         "datasource_name": "items",
+         "datasource_loop_variable": "name"
+      }
+}
+   ```
+   In the json above, the `input_property` key holds the property that is to be validated before the request to that path is passed. The `datasource_name` key holds the name of the datasource(a list) from the database. The `datasource_loop_variable` key holds the name of the key on each object in the datasource.
+
+   The rego rules combine data from the database with the input object, to work out certain conditions. The datasource is the list of values from the database, and the datasource_loop_variable is the name of the key on each object in the datasource.
```
Comment (Youngestdev) [inline docs.md:265]:
```suggestion
   The rego rules combine data from the database with the input object, to work out certain conditions. The datasource is the list of values from the database, and the `datasource_loop_variable` is the name of the key on each object in the datasource.
```

File: docs.md [hunk 43]
```diff
@@ -1,23 +1,499 @@
...
+   ```json
+ {
+      "command": "input_prop_in_as",
+      "properties": {
+         "datasource_name": "items",
+         "datasource_loop_variables": ["name", "everyone"],
+         "input_properties": ["preferred_username", "groupname"]
+      }
+ }
+   ```
```
Comment (Youngestdev) [inline docs.md:285]: Format this section properly.

File: docs.md [hunk 44]
```diff
@@ -1,23 +1,499 @@
...
+      input.groupname == "EDITOR_ATAC"
+   }
+   ```
+
+
+
+
+POST `/policies/`  Create new policy
+============
+Using the JSON defined from the frontend, this route is used to build a new policy. The response will be a REGO file written and pushed to GitHub as a newly established remote repository, with the request body conforming to a specified syntax described by the pydantic model `Policy`. <br />
```
Comment (Youngestdev) [inline docs.md:324]: Again, we should not make it seem like only the frontend can send JSON. A better way to start this section:

The POST route takes a request body containing the rules defined according to the schema.

**Review by Youngestdev — COMMENTED** [review]
Look good. Left a few comments.

File: docs.md [hunk 45]
```diff
@@ -1,23 +1,499 @@
...
+```
+In the json above, the `value` key holds the value that is to be checked for equality, while the `input_property` key holds the property that is to be checked in the input object.
+
+The REGO equivalent of the above rule object is: <br />
+```rego
+input.company == "Geobeyond srl"
+```
+
+
+
```
Comment (Youngestdev) [inline docs.md:217]:
```suggestion
```

File: docs.md [hunk 46]
```diff
@@ -1,23 +1,499 @@
...
+```
+In the JSON above, the list of objects indicates a very special `allow` block, that combines two commands. The `input_property` key holds the property that is to be validated before the request to that path is passed. 
+
+The REGO equivalent of the above rule object is: <br />
+```rego
+allow {
+   input.request_path == ["v1", "collections", "lakes", ""]
+   input.groupname == "admin"
+}
+```
```
Comment (Youngestdev) [inline docs.md:248]:
```suggestion
```

```

File: docs.md [hunk 47]
```diff
@@ -1,23 +1,499 @@
...
+In the json above, the `input_property` key holds the property that is to be validated before the request to that path is passed. The `datasource_name` key holds the name of the datasource(a list) from the database. The `datasource_loop_variable` key holds the name of the key on each object in the datasource.
+
+The rego rules combine data from the database with the input object, to work out certain conditions. The datasource is the list of values from the database, and the `datasource_loop_variable` is the name of the key on each object in the datasource.
+
+The REGO equivalent of the above rule object is: <br />
+```rego
+input.company == data.items[i].name
+```
+### input_props_in_as
```
Comment (Youngestdev) [inline docs.md:272]:
```suggestion

### input_props_in_as
```

File: docs.md [hunk 48]
```diff
@@ -1,23 +1,499 @@
...
+{
+   "command": "input_prop_in_as",
+   "properties": {
+      "datasource_name": "items",
+      "datasource_loop_variables": ["name", "everyone"],
+      "input_properties": ["preferred_username", "groupname"]
+   }
+}
+```
+In the json above, the resulting REGO code loops over the datasource twice, checking for equality between the values of the input properties, and the values of the datasource loop variables.
```
Comment (Youngestdev) [inline docs.md:286]:
```suggestion

In the JSON above, the resulting REGO code loops over the datasource twice, checking for equality between the values of the input properties, and the values of the datasource loop variables.
```

File: docs.md [hunk 49]
```diff
@@ -1,23 +1,499 @@
...
+
+```rego
+allow {
+   input.groupname == "EDITOR_ATAC"
+}
+```
+
+
+
+
```
Comment (Youngestdev) [inline docs.md:321]:
```suggestion
```
